fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -345,6 +345,129 @@ public FDBRecordContext openContext(@Nullable Map<String, String> mdcContext,
+     *
+     * @return a future that will complete after being run by the FDB network thread
+     * @see #performNoOp(Map, FDBStoreTimer)
+     */
+    @Nonnull
+    public CompletableFuture<Void> performNoOpAsync() {
+        return performNoOpAsync(null);
+    }
+
+    /**
+     * Perform a no-op against FDB to check network thread liveness. See {@link #performNoOp(Map, FDBStoreTimer)}
+     * for more information. This will use the default MDC for running threads.
+     *
+     * @param timer the timer to use for instrumentation
+     * @return a future that will complete after being run by the FDB network thread
+     * @see #performNoOp(Map, FDBStoreTimer)
+     */
+    @Nonnull
+    public CompletableFuture<Void> performNoOpAsync(@Nullable FDBStoreTimer timer) {
+        return performNoOpAsync(null, timer);
+    }
+
+    /**
+     * Perform a no-op against FDB to check network thread liveness. This operation will not change the underlying data
+     * in any way, nor will it perform any I/O against the FDB cluster. However, it will schedule some amount of work
+     * onto the FDB client and wait for it to complete. The FoundationDB client operates by scheduling onto an event
+     * queue that is then processed by a single thread (the "network thread"). This method can be used to determine if
+     * the network thread has entered a state where it is no longer processing requests or if its time to process
+     * requests has increased. If the network thread is busy, this operation may take some amount of time to complete,
+     * which is why this operation returns a future.
+     *
+     * <p>
+     * If the provided {@link FDBStoreTimer} is not {@code null}, then this will update the {@link FDBStoreTimer.Events#PERFORM_NO_OP}
+     * operation with related timing information. This can then be monitored to detect instances of client saturation
+     * where the performance bottleneck lies in scheduling work to run on the FDB event queue.
+     * </p>
+     *
+     * @param mdcContext logger context to set in running threads
+     * @param timer the timer to use for instrumentation
+     * @return a future that will complete after being run by the FDB network thread
+     */
+    @Nonnull
+    public CompletableFuture<Void> performNoOpAsync(@Nullable Map<String, String> mdcContext,
+                                                    @Nullable FDBStoreTimer timer) {
+        final FDBRecordContext context = openContext(mdcContext, timer);
+        boolean futureStarted = false;
+
+        try {
+            // Set the read version of the transaction, then read it back. This requires no I/O, but it does
+            // require the network thread be running. The exact value used for the read version is unimportant.
+            final Transaction tr = context.ensureActive();
+            final long startTime = System.nanoTime();
+            tr.setReadVersion(1066L);
+            CompletableFuture<Long> future = tr.getReadVersion();
```

Comment:
I debated with using the `asyncToSync` timeout here with the `PERFORM_NO_OP`, but I decided against it as (1) that won't actually cancel the future anyway and (2) this allows the instrumentation to be on the full wait (if it so happens to exceed the timeout value set). But perhaps that is wrong.